### PR TITLE
setted the localstorage for fonts, name, font-sizes and a background

### PR DESCRIPTION
**Pull request by vasystus**

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
 import Layout from "../components/Layout";
+import useLocalStorageState from "../components/Hooks/useLocalStorageState";
+
 function MyApp({ Component, pageProps }) {
+  // Custom hooks setting user user details - here in the _app.js so it can be accessed acrosss all componenets!
+  const [learnerName, setLearnerName] = useLocalStorageState(
+    "learner-name",
+    null
+  ) || [null, null];
+  const [font, setFont] = useLocalStorageState("font", null) || [null, null];
```
Comment (PJSalter) [inline pages/_app.js]: I really like the use of the local storage use state taking an initial statement to null, once a font is selected then there will be a selected for a specific learner ✨

File: pages/_app.js [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
+  ) || [null, null];
+  const [background, setBackground] = useLocalStorageState(
+    "background",
+    null
+  ) || [null, null];
+
   return (
     <Layout>
-      <Component {...pageProps} />
+      <Component
```
Comment (PJSalter) [inline pages/_app.js]: Great work on the specific route once learner navigates, the props become active and set these components to become active ✨

**Review by joe-dev-public — APPROVED** [review]

File: pages/child-landing.js [hunk 3]
```diff
@@ -2,10 +2,10 @@ import Tts from "../components/tts";
 import Link from "next/link";
 import styled from "styled-components";
 
-export default function ChildLanding() {
+export default function ChildLanding({ learnerName }) {
   return (
     <>
-      <h2>Hi, --learners name here, which we can get from localStorage --!</h2>
+      <h2>{learnerName ? `Hi, ${learnerName}!` : "Hi!"}!</h2>
```
Comment (PJSalter) [inline pages/child-landing.js:8]: changed the string text to hi instead of little learner, well done

File: pages/learner-onboarding/name.js [hunk 4]
```diff
@@ -1,27 +1,24 @@
...
-// 5. orange next button on bottom center.
-// 6. back button on bottom left.
-
 import styled from "styled-components";
 import LearnerOnboardingLayout from "../../components/LearnerOnboardingLayout";
 
-export default function LearnerChoiceName() {
+export default function LearnerChoiceName( {learnerName, setLearnerName}) {
+
+  const saveName = (e) => {
```
Comment (PJSalter) [inline pages/learner-onboarding/name.js:6]: awesome now we can save the name of the user, awesome work on this function

File: pages/learner-onboarding/name.js [hunk 5]
```diff
@@ -1,27 +1,24 @@
...
       stepNumber={1}
       nextStep="font-selection"
       previousStep="avatar-selection"
     >
-      {/* <LearnerOnboardingName /> */}
       <StyledForm>
-        <label for="learnerName"></label>
-        <StyledInput type="text" name="learnerName" />
+        <label htmlFor="learnerName"></label>
+        <StyledInput onChange={saveName} type="text" name="learnerName" />
```
Comment (PJSalter) [inline pages/learner-onboarding/name.js:20]: good use with onChange, once user types there name

**Review by PJSalter — APPROVED** [review]

**Pull request merged**